src/INode.java

```diff
@@ -174,9 +175,23 @@ public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException
 		}
 	}
 
-	private ASTValue evalAdd(WordsEnvironment environment) {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+	private ASTValue evalAdd(WordsEnvironment environment) throws WordsRuntimeException {
+		ASTValue lhs = children.get(0).eval(environment);
+		ASTValue rhs = children.get(1).eval(environment);
+		
+		if(((lhs.type == AST.ValueType.STRING) && (rhs.type == AST.ValueType.NUM)) || 
+				((lhs.type == AST.ValueType.NUM) && (rhs.type == AST.ValueType.STRING)) ||
+				((lhs.type == AST.ValueType.STRING) && (rhs.type == AST.ValueType.STRING))) {
+			lhs.tryCoerceTo(AST.ValueType.STRING);
+			rhs.tryCoerceTo(AST.ValueType.STRING);
+			return new ASTValue(lhs.stringValue+rhs.stringValue);
+		}
+		else if((lhs.type == AST.ValueType.NUM) && (rhs.type == AST.ValueType.NUM)) {
```

Comment:
We should follow our style guide, which means the else should be on the same line as the brace, i.e., 
if {
   ...
} else if {
   ...
} else {
   ...
}